### PR TITLE
Use the initial `originalQuestion` in the `saveModal`

### LINKED ISSUE #42773: "-modified" suffix briefly shown in a model's name input during creation
### Describe the bug

https://github.com/metabase/metabase/assets/6830683/1bba1142-2652-4ed0-adf8-9834a56457ef

### To Reproduce

1. Start a new notebook model based on Orders table
2. Click "Save"
3. Observe name input in the modal from now
4. Click "Save" in the modal

Observe that the name input will briefly show `- modified` suffix

### Expected behavior

`- modified` suffix is not shown

### Information about your Metabase installation

master, 44cb385c40

### Severity

P3

### PR DESCRIPTION
**Pull request by nemanjaglumac**
### Description
This PR fixes #42773 by mimicking the solution we already have in `master` (that's proven to be working).

tl;dr
We need to remember the *initial* state of the "original" question because it changes to the ad-hoc question in Redux just before the save modal is closed.

### How to verify
Follow the repro steps and make sure you don't see `- Modified` at any stage during the question/metric/model save.

### Checklist
- [x] Tests have been added/updated to cover changes in this PR

> [!Important]
> Adds the same repro from #47103

### CODE REVIEW

**Review by romeovs — APPROVED** [review]
Fix works and code makes sense.
